### PR TITLE
Fix generation with multi apiversion

### PR DESCRIPTION
**Pull request by davidfestal**
### What does this PR do?

This PR fixes the code / CRD and schema generation process to make it compatible with the support of several K8S API versions of the `workspace.devfile.io` api group. 

### What issues does this PR fix or reference?

This will fix issue https://github.com/devfile/api/issues/247 and is somehow a prerequisite to issue https://github.com/devfile/api/issues/150

### Is your PR tested? Consider putting some instruction how to test your changes

Yes, it was tested.

To test it, just change (temporarily) some source files of the `v1alpha1` package and run the generation:

- Add the following line just above the `package v1alpha1` line in file `pkg/apis/workspaces/v1alpha1/doc.go` file:
```go
// +devfile:jsonschema:version=2.0.0-alpha1
```
- Add the following line just above the `type DevWorkspaceTemplateSpec struct {` line in file `pkg/apis/workspaces/v1alpha1/devworkspaceTemplateSpec.go` file:
```go
// +devfile:jsonschema:generate
```
- Generate the Json Schemas with the `./build.sh` command

- Observe the new schemas created in the `v1alpha1` schema sub-folder.

If you update your previous change from: 
```go
// +devfile:jsonschema:version=2.0.0-alpha1
```
to
```go
// +devfile:jsonschema:version=2.0.0-beta1
```
And run the generation again, you will get the following expected error:
```
Generating JsonSchemas
github.com/devfile/api/pkg/apis/workspaces/v1alpha2:-: The schema versions should be incremented on each increment of the corresponding K8S apiVersion.
This is not the case in the 'workspace.devfile.io' API group:
  'v1alpha1' K8S apiVersion => '2.0.0-beta1' Json schema version
  'v1alpha2' K8S apiVersion => '2.0.0-alpha2' Json schema version

Error: not all generators ran successfully
```

### CODE REVIEW

File: generator/genutils/kube_liike_versions.go
Comment (GeekArthur) [inline generator/genutils/kube_liike_versions.go:39]: Nit: should we directly use `SortKubeLikeVersion(versionsToSort)` here?

**Review by GeekArthur — APPROVED** [review]
Thanks for making the changes, it resolves the CRD generation issue that I open. The code LGTM and I just leave a nit comment.

**Review by johnmcollier — COMMENTED** [review]
Should we add `// +devfile:jsonschema:generate` to `pkg/apis/workspaces/<version>/devworkspaceTemplateSpec.go`? For v1alpha1 and v1alpha2.

File: generator/genutils/kube_liike_versions.go
Comment (amisevsk) [inline generator/genutils/kube_liike_versions.go:1]: Typo in this file name (`liike -> like`)

File: generator/genutils/kube_liike_versions.go
Comment (amisevsk) [inline generator/genutils/kube_liike_versions.go:33]: Does it make sense to instead walk the list and compare using `version.CompareKubeAwareVersionStrings`? e.g.
```go
latest := versions[0]
for _, ver := range versions {
  if version.CompareKubeAwareVersionStrings(ver, latest) > 0 {
    latest = ver
  }
}
return ver
```

File: generator/schemas/gen.go
Comment (amisevsk) [inline generator/schemas/gen.go:187]: If we're using `fmt.Errorf` we should use formatting rather than string concatenation:
```suggestion
					packageByGV[groupVersion].AddError(fmt.Errorf("The schema versions should be incremented on each increment of the corresponding K8S apiVersion.\n" +
						"This is not the case in the "%s' API group:\n" +
						"  '%s' K8S apiVersion => '%s' Json schema version\n" +
						"  '%s' K8S apiVersion => '%s' Json schema version\n", groupName, currentAPIVersion, currentSchemaVersion.String(), apiVersion, schemaVersion.String()))
```
Consider also using a raw string to avoid having to include `\n" +`

**Review by amisevsk — APPROVED** [review]
A couple of minor comments in addition to the others. Approving so as to not block merge.

File: generator/genutils/kube_liike_versions.go
Comment (davidfestal) [reply generator/genutils/kube_liike_versions.go:39]: You're right. I'll fix it

File: generator/genutils/kube_liike_versions.go
Comment (davidfestal) [reply generator/genutils/kube_liike_versions.go:33]: Sure, it certainly makes sense. I assume I was just lazy and thus reused sort.

File: generator/genutils/kube_liike_versions.go
Comment (davidfestal) [reply generator/genutils/kube_liike_versions.go]: Fixed by applying @amisevsk suggestion in [this comment](https://github.com/devfile/api/pull/273#discussion_r539852577)

File: generator/genutils/kube_liike_versions.go
Comment (davidfestal) [reply generator/genutils/kube_liike_versions.go]: Fixed in commit https://github.com/devfile/api/pull/273/commits/b65067d128173c58b17a9c2d908743e95e901b7e

File: generator/genutils/kube_liike_versions.go
Comment (davidfestal) [reply generator/genutils/kube_liike_versions.go]: Fixed in commit b65067d

**Comment by davidfestal** [comment]
> devworkspaceTemplateSpec

Why would it be necessary ?

**Pull request merged**
